### PR TITLE
Feat: slices/charts rbac and access the explore for viz of chart.

### PR DESCRIPTION
**Pull request by nikhil-kuyya**
### SUMMARY
Chart control based on role based access

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [X] Required feature flags:
- [X] Changes UI
- [X] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [X] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/charts/filters.py [hunk 1]
```diff
@@ -80,3 +84,106 @@ def apply(self, query: Query, value: Any) -> Query:
...
+            db.session.query(Slice.id)
+            .join(Slice.owners)
+            .filter(
+                security_manager.user_model.id 
+                == security_manager.user_model.get_user_id()
+            )
+        )
+
+        # favorited Slices for the user
+        users_favorite_slice_query = db.session.query(FavStar.obj_id).filter(
```
Comment (nikhil-kuyya) [inline superset/charts/filters.py:157]: favorite for filter. can remove from permission.
check the scenario of update/moving the ownership.

File: superset/charts/filters.py [hunk 2]
```diff
@@ -80,3 +84,106 @@ def apply(self, query: Query, value: Any) -> Query:
...
+                )
+            )
+            feature_flagged_filters.append(Slice.id.in_(roles_based_query))
+
+
+        query = query.filter(
+            or_(
+                Slice.id.in_(owner_ids_query),
+                Slice.id.in_(datasource_perm_query),
+                Slice.id.in_(users_favorite_slice_query),
```
Comment (nikhil-kuyya) [inline superset/charts/filters.py:184]: check the above fav. scenario and change it.

**Review by meyyappan66 — APPROVED** [review]

File: superset/charts/data/commands/get_data_command.py [hunk 3]
```diff
@@ -61,4 +65,16 @@ def run(self, **kwargs: Any) -> Dict[str, Any]:
...
 
     def validate(self) -> None:
-        self._query_context.raise_for_access()
+        """
+         Custom Access provision.
+         Intial Query Context might be looking for data sources access in permissions.
+         now we have implemented to look for the custom access for owners and add more details.
+        
+        """
+        if is_feature_enabled("CHART_RBAC"):
```
Comment (nikhil-kuyya) [inline superset/charts/data/commands/get_data_command.py:74]: add else for main if condition.

File: superset/charts/filters.py (hunk 1 above)
Comment (nikhil-kuyya) [reply superset/charts/filters.py:157]: it is not clearning the favorite record if the permission/ownership is removed.
So user still able to access the slice.
Which also not required so removing the filter conditon on this.

File: superset/charts/filters.py (hunk 2 above)
Comment (nikhil-kuyya) [reply superset/charts/filters.py:184]: if we allow the slice here but not allow iat security manger by favstar will cause issue.
so removing the favstar as base filter.

Comment (nikhil-kuyya) [reply superset/charts/filters.py:184]: as it might be required to get the chart but the filter condition.

**Review by nikhil-kuyya — COMMENTED** [review]
as per request changes are made. pushing to server to test more.

**Pull request merged**